### PR TITLE
[REVIEW] add ability to query if PTDS is enabled

### LINKED ISSUE #6111: [FEA] add API to libcudf so we can query at runtime whether it was compiled for PTDS
When running the RAPIDS spark plugin, we'd like to be able to log if PTDS is on or off in our environment.

In order to do this, we'll need to query libcudf at runtime to know if it's compiled for PTDS or not. We'd like to propose some static variable defined if PTDS is on so we can inspect it at runtime. Something like:

```
#ifdef CUDA_API_PER_THREAD_DEFAULT_STREAM
bool cudf::ptds_enabled = true;
#else
bool cudf::ptds_enabled = false;
#endif
```

Not sure where it should go or if the above make sense, but something to that effect, so we can include a header in the jni side and ask for this to check for our log.

### PR DESCRIPTION
**Pull request by rongou**
Closes #6111.

This is an alternative, simpler solution than #6324. We can revisit this once https://github.com/rapidsai/rmm/pull/413 is merged.

@harrism @jrhemstad @kkraus14 @abellina

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Review by raydouglass — APPROVED** [review]

File: cpp/include/cudf/utilities/default_stream.hpp [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * limitations under the License.
+ */
+
+#pragma once
+
+namespace cudf {
+namespace default_stream {
+
+/// Whether the default stream is per-thread.
+extern bool const is_per_thread;
```
Comment (davidwendt) [inline cpp/include/cudf/utilities/default_stream.hpp:23]: I would recommend making this a function and to not introduce a new namespace for this
```
bool is_per_thread_default_stream();
```

Comment (rongou) [reply cpp/include/cudf/utilities/default_stream.hpp]: Done.

**Review by harrism — COMMENTED** [review]
Since this takes no arguments and is not a method on an object, I would add _enabled to the end of the function name.

File: cpp/include/cudf/utilities/default_stream.hpp [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+#pragma once
+
+namespace cudf {
+
+/**
+ * @brief Check if the default stream is per-thread.
+ *
+ * @return true if the default stream is per-thread, false otherwise.
+ */
+bool is_per_thread_default_stream();
```
Comment (devavret) [inline cpp/include/cudf/utilities/default_stream.hpp:26]: What do folks think about `is_ptds_enabled()`?

Comment (rgsl888prabhu) [reply cpp/include/cudf/utilities/default_stream.hpp:26]: I like it

Comment (rongou) [reply cpp/include/cudf/utilities/default_stream.hpp]: Done.

**Review by devavret — APPROVED** [review]

**Comment by rongou** [comment]
rerun tests

**Review by abellina — APPROVED** [review]

**Review by harrism — APPROVED** [review]

**Comment by rgsl888prabhu** [comment]
rerun tests

**Comment by rongou** [comment]
rerun tests

**Pull request merged**